oshmem/mca/memheap/base/memheap_base_mkey.c

```diff
@@ -445,14 +445,14 @@ void memheap_oob_destruct(void)
 static int send_buffer(int pe, pmix_data_buffer_t *msg)
 {
     void *buffer;
-    int32_t size;
+    size_t size;
     int rc;
 
     PMIX_DATA_BUFFER_UNLOAD(msg, buffer, size);
     rc = PMPI_Send(buffer, size, MPI_BYTE, pe, 0, oshmem_comm_world);
     free(buffer);
 
-    MEMHEAP_VERBOSE(5, "message sent: dst=%d, rc=%d, %d bytes!", pe, rc, size);
+    MEMHEAP_VERBOSE(5, "message sent: dst=%d, rc=%d, %zu bytes!", pe, rc, size);
```

Comment:
Please use the appropriate `PRI` printf code for portability: https://github.com/open-mpi/ompi/wiki/PrintfCodes